saleor/discount/utils.py

```diff
@@ -55,6 +67,54 @@
 
 CatalogueInfo = defaultdict[str, set[Union[int, str]]]
 CATALOGUE_FIELDS = ["categories", "collections", "products", "variants"]
+T_Model = TypeVar("T_Model", bound=type[Union[Checkout, Order]])
+T_LineModel = TypeVar("T_LineModel", bound=type[Union[CheckoutLine, OrderLine]])
+T_DiscountModel = TypeVar(
+    "T_DiscountModel", bound=type[Union[CheckoutDiscount, OrderDiscount]]
+)
+T_LineDiscountModel = TypeVar(
+    "T_LineDiscountModel", bound=type[Union[CheckoutLineDiscount, OrderLineDiscount]]
+)
+T_LineInfo = TypeVar(
+    "T_LineInfo", bound=type[Union[CheckoutLineInfo, DraftOrderLineInfo]]
+)
+
+
+@dataclass
+class CheckoutOrOrderHelper(
```

Comment:
IMO we should find out the better name for that. Something connected to discount, maybe `DiscountData`?

Reply:
I got rid of the helper.